common/src/main/java/net/william278/husktowns/hook/Pl3xMapHook.java

```diff
@@ -0,0 +1,147 @@
+package net.william278.husktowns.hook;
+
+import net.pl3x.map.core.Pl3xMap;
+import net.pl3x.map.core.event.EventHandler;
+import net.pl3x.map.core.event.EventListener;
+import net.pl3x.map.core.event.server.Pl3xMapEnabledEvent;
+import net.pl3x.map.core.event.world.WorldLoadedEvent;
+import net.pl3x.map.core.event.world.WorldUnloadedEvent;
+import net.pl3x.map.core.markers.Point;
+import net.pl3x.map.core.markers.layer.SimpleLayer;
+import net.pl3x.map.core.markers.marker.Marker;
+import net.pl3x.map.core.markers.option.Options;
+import net.pl3x.map.core.markers.option.Tooltip;
+import net.william278.husktowns.HuskTowns;
+import net.william278.husktowns.claim.TownClaim;
+import net.william278.husktowns.claim.World;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Collection;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.logging.Level;
+import java.util.stream.Collectors;
+
+public class Pl3xMapHook extends MapHook implements EventListener {
+
+    private static final String CLAIMS_LAYER = "claim_markers";
+    private final ConcurrentLinkedQueue<TownClaim> claims = new ConcurrentLinkedQueue<>();
```

Comment:
Instead of a ConcurrentLinkedQueue here, I'd do a ConcurrentHashMap of ClaimWorld<->TownClaims, as this is the data structure as stored by the plugin. Would make things a tad more complex, but probably necessary to filter by world

Reply:
Resolved in [624f586](https://github.com/WiIIiam278/HuskTowns/pull/303/commits/624f58658bb6397408843cb1c7cf439b9782809a)